src/crawl/core_types.rs

```diff
@@ -36,19 +41,29 @@ pub struct Crawler {
 
 impl CrawledNode {
     /// Create a new CrawledNode using its hostname, connectivity status and Qset.
-    pub(crate) fn new(url: String, online: bool, quorum_set: McQuorumSet) -> Self {
+    pub(crate) fn new(
+        url: String,
+        online: bool,
+        quorum_set: McQuorumSet,
+        latest_ledger: u64,
+        network_block_version: u32,
+        minimum_fee: u64,
+    ) -> Self {
         let (domain, port) = Self::fragment_mc_url(url);
         CrawledNode {
             public_key: Ed25519Public::default(),
             domain,
             port,
             quorum_set,
             online,
+            latest_ledger,
+            network_block_version,
+            minimum_fee,
         }
     }
 
     /// Return 0.0.0.0 as an address if not resolvable otherwise the stats functions would return one own's geolocation
-    fn fragment_mc_url(url: String) -> (String, u16) {
+    pub fn fragment_mc_url(url: String) -> (String, u16) {
```

Comment:
Yes, it has to be because we use `fragment_mc_url()` in `report.rs` when checking for trusted nodes.
If the function wasn't public, it would be out of scope.

Reply:
``` pub (crate) ``` should suffice.

Reply:
alright